### PR TITLE
Adding clang format check

### PR DESCRIPTION
**Pull request by mwestphal**

### CODE REVIEW

**Comment by mwestphal** [comment]
related to #194 

I will merge this only when there is no ongoing PR.

**Comment by Meakk** [comment]
Can you make the github actions push a formatting commit automatically?  
I guess it would be difficult since we need author's fork write access...

**Comment by mwestphal** [comment]
Not supported by the action and I dont like a lot this kind of stuff. 

This would of course be doable using tokens and such.

Running a command locally is easy enough for most people I hope. Ideally, I would like to give the command in a comment but could not find how to do it.

**Comment by mwestphal** [comment]
@Meakk ready for review, I did not actually include .in file as excluding @foo@ var was a pain.

File: src/winshellext/F3DThumbnailProvider.h [hunk 1]
```diff
@@ -42,6 +41,8 @@ class F3DThumbnailProvider
 
 private:
   long m_cRef;
-  wchar_t m_f3dPath[MAX_PATH];  // The path to f3d executable that will be used to produce the thumbnail
-  wchar_t m_filePath[MAX_PATH]; // The path to the file for which we will have to produce the thumbnail
+  wchar_t
```
Comment (Meakk) [inline src/winshellext/F3DThumbnailProvider.h:44]: maybe move the comment to the previous line to avoid this weird indentation

**Review by Meakk — APPROVED** [review]
minor comment, then it's good to go

**Pull request merged**